config-overrides.js

```diff
@@ -1,45 +1,73 @@
+          // See: https://github.com/vercel/next.js/pull/9012
+          framework: {
+            name: 'framework',
+            test: /(?<!node_modules.*)[\\/]node_modules[\\/](react|react-dom|scheduler|use-subscription|styled-components)[\\/]/,
+          },
+          coreWeb3: {
+            name: 'core-web3',
+            test: /(?<!node_modules.*)[\\/]node_modules[\\/](@ethersproject|bn\.js|aes-js)[\\/]/,
+          },
+        },
+      };
+    }
 
-  // See: https://www.alchemy.com/blog/how-to-polyfill-node-core-modules-in-webpack-5
-  config.resolve.fallback = {
-    ...config.resolve.fallback,
-    http: require.resolve('stream-http'),
-    https: require.resolve('https-browserify'),
-    os: require.resolve('os-browserify'),
-    stream: require.resolve('stream-browserify'),
-  };
+    // See: https://www.alchemy.com/blog/how-to-polyfill-node-core-modules-in-webpack-5
+    config.resolve.fallback = {
+      ...config.resolve.fallback,
+      http: require.resolve('stream-http'),
+      https: require.resolve('https-browserify'),
+      os: require.resolve('os-browserify'),
+      stream: require.resolve('stream-browserify'),
+    };
 
-  config.plugins = (config.plugins || []).concat([
-    new webpack.ProvidePlugin({
-      Buffer: ['buffer', 'Buffer'],
-    }),
-  ]);
+    config.plugins = (config.plugins || []).concat([
+      new webpack.ProvidePlugin({
+        Buffer: ['buffer', 'Buffer'],
+      }),
+      // The @web3modal/react package spawns too many async chunks, so we limit the total amount of chunks
+      new webpack.optimize.LimitChunkCountPlugin({
+        maxChunks: 5,
+      }),
+    ]);
 
-  config.ignoreWarnings = [
-    // Ignore warnings raised by source-map-loader for third party packages.
-    // See: https://github.com/facebook/create-react-app/pull/11752#issuecomment-1137238078
-    /**
-     *
-     * @param {import('webpack').WebpackError} warning
-     * @returns {boolean}
-     */
-    function ignoreSourceMapLoaderWarnings(warning) {
-      return (
-        warning.module &&
-        warning.module.resource.includes('node_modules') &&
-        warning.details &&
-        warning.details.includes('source-map-loader')
-      );
-    },
-  ];
+    config.ignoreWarnings = [
+      // Ignore warnings raised by source-map-loader for third party packages.
+      // See: https://github.com/facebook/create-react-app/pull/11752#issuecomment-1137238078
+      /**
+       *
+       * @param {import('webpack').WebpackError} warning
+       * @returns {boolean}
+       */
+      function ignoreSourceMapLoaderWarnings(warning) {
+        return (
+          warning.module &&
+          warning.module.resource.includes('node_modules') &&
+          warning.details &&
+          warning.details.includes('source-map-loader')
+        );
+      },
+    ];
 
-  return config;
+    return config;
+  },
+
+  jest(config) {
+    config.transformIgnorePatterns = [
```

Comment:
nit: I'd add an assert that verifies `transformIgnorePatterns` is what it is now. Since you are overriding the values, it might be pretty hard to debug in the future in case they modify its content.